### PR TITLE
LPS-153508 | Test Automation for LPS-94874: Can't edit segment if the segment has a select input

### PR DESCRIPTION
**Pull request by rafaellajordao**
Hi! :)
This pr is corresponding to this ticket: [LPS-94874](https://issues.liferay.com/browse/LPS-94874) -> [LPS-153508](https://issues.liferay.com/browse/LPS-153508)

### CODE REVIEW

**Comment by rafaellajordao** [comment]
ci:test:sf

**Comment by rafaellajordao** [comment]
ci:test:tango

**Comment by rafaellajordao** [comment]
> ###  ci:test:tango - 158 out of 175 jobs passed in 2 hours 16 minutes
> Click here for more details.

these failures have nothing to do with the test correction :)

**Comment by Luizcarlosqueiroz** [comment]
Hey @rafaellajordao! I think there's only need to go to the Segment and check the toggle button. Because in this bug, when the user goes to edit, the "edit toggle" is disabled.
disable-toggle

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationEditSegment.testcase [hunk 1]
```diff
@@ -36,6 +36,55 @@ definition {
...
+
+			Segmentation.fillProperty(selectCheckbox = "Power User");
+
+			Button.clickSave();
+
+			Segmentation.editSegment(entry = "EditSegmentIfHaveASelectInput Test");
+
+			AssertChecked.assertCheckedNotVisible(
+				checkboxName = "",
+				locator1 = "Checkbox#ANY_CHECKBOX_CHECKED");
```
Comment (rafaellajordao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationEditSegment.testcase:80]: hi @Luizcarlosqueiroz !! :)

I added this new step to the tests to check the toggle, now it's more complete!! What do you think? 

Or would you prefer that I just leave this step in the entire test?

**Comment by rafaellajordao** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationEditSegment.testcase (hunk 1 above)
Comment (Luizcarlosqueiroz) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationEditSegment.testcase:80]: Hey Rafa!
I think you could "merge" both steps I've created. For example, when user goes to edit segment, the toggle is enabled and checked (and user can edit it), but when the toggle is enabled and unchecked, the user can't edit the segment. Do you think this covers the scenario well?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationEditSegment.testcase [hunk 2]
```diff
@@ -36,6 +36,55 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-94874."
+	@priority = "5"
+	@uitest
```
Comment (Luizcarlosqueiroz) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationEditSegment.testcase:41]: Nice use of the tag!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationEditSegment.testcase [hunk 3]
```diff
@@ -36,6 +36,55 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-94874."
```
Comment (Luizcarlosqueiroz) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationEditSegment.testcase:39]: You could add the bug description here, like this:
`@description = "LPS-151808 Bug: The button Metrics is shown for web content in Pending status"`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationEditSegment.testcase (hunk 1 above)
Comment (rafaellajordao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationEditSegment.testcase:80]: i got it!!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationEditSegment.testcase (hunk 3 above)
Comment (rafaellajordao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationEditSegment.testcase:39]: ok!!

**Comment by rafaellajordao** [comment]
ci:test:sf

**Review by Luizcarlosqueiroz — APPROVED** [review]
Good to go! :smile:

**Comment by Luizcarlosqueiroz** [comment]
ci:test:tango

**Comment by rafaellajordao** [comment]
> these failures have nothing to do with the test correction :)

these failures have nothing to do with the test correction :)

**Comment by rafaellajordao** [comment]
ci:forward

**Pull request closed without merging**